packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx

```diff
@@ -101,7 +102,25 @@ function RoutingForm({ form, profile, ...restProps }: Props) {
       if (decidedAction.type === "customPageMessage") {
         setCustomPageMessage(decidedAction.value);
       } else if (decidedAction.type === "eventTypeRedirectUrl") {
-        await router.push(`/${decidedAction.value}?${allURLSearchParams}`);
+        const regex = /\{([^\}]+)\}/g;
+
+        const variables: string[] =
+          decidedAction.value.match(regex)?.map((match: string) => match.slice(1, -1)) || [];
+
+        let eventTypeUrl = decidedAction.value;
+
+        variables.forEach((variable) => {
+          for (const key in response) {
+            const identifier = getFieldIdentifier(fields.find((field) => field.id === key));
+            if (identifier === variable) {
+              eventTypeUrl = eventTypeUrl.replace(
+                `{${variable}}`,
+                slugify(response[key].value.toString() || "")
+              );
+            }
+          }
+        });
+        await router.push(`/${eventTypeUrl}?${allURLSearchParams}`);
```

Comment:
We happened  to change the same piece of code for 2 different reasons  😅

Reply:
yeah I saw that too 😅 we can merge yours first and I'll handle the conflicts

Reply:
Mine is not that urgent. Happy to merge it first. Could you extract out this logic in a separate function

Reply:
extracted it into a separate function, @hariombalhara can you re-review?